src/extractor/intersection/intersection_analysis.cpp

```diff
@@ -0,0 +1,808 @@
+    EdgeID roundabout_from = SPECIAL_EDGEID, roundabout_to = SPECIAL_EDGEID;
+    double roundabout_from_angle = 0., roundabout_to_angle = 0.;
+
+    for (const auto eid : graph.GetAdjacentEdgeRange(intersection_node))
+    {
+        const auto &edge_data = graph.GetEdgeData(eid);
+        const auto &edge_class = edge_data.flags;
+        const auto to_node = graph.GetTarget(eid);
+        const auto reverse_edge = graph.FindEdge(to_node, intersection_node);
+        BOOST_ASSERT(reverse_edge != SPECIAL_EDGEID);
+
+        const auto is_exit_edge = !edge_data.reversed && !isTurnRestricted(restrictions, to_node);
+        const auto is_bidirectional = !graph.GetEdgeData(reverse_edge).reversed;
+        allowed_exits += is_exit_edge;
+        bidirectional_edges += is_bidirectional;
+
+        if (edge_class.roundabout || edge_class.circular)
+        {
+            if (edge_data.reversed)
+            {
+                // "Linked Roundabouts" is an example of tie between two linked roundabouts
+                // A tie breaker for that maximizes ∠(roundabout_from_bearing, ¬from_edge_bearing)
+                const auto angle = util::bearing::angleBetween(
+                    findEdgeBearing(geometries, reverse_edge), from_edge_reversed_bearing);
+                if (angle > roundabout_from_angle)
+                {
+                    roundabout_from = reverse_edge;
+                    roundabout_from_angle = angle;
+                }
+            }
+            else
+            {
+                // a tie breaker that maximizes ∠(¬from_edge_bearing, roundabout_to_bearing)
+                const auto angle = util::bearing::angleBetween(from_edge_reversed_bearing,
+                                                               findEdgeBearing(geometries, eid));
+                if (angle > roundabout_to_angle)
+                {
+                    roundabout_to = eid;
+                    roundabout_to_angle = angle;
+                }
+            }
+        }
+    }
+
+    // 3) if the intersection has a barrier
+    const bool is_barrier_node = barrier_nodes.find(intersection_node) != barrier_nodes.end();
+
+    // Check a U-turn
+    if (from.node == destination_node)
+    {
+        // Allow U-turns before barrier nodes
+        if (is_barrier_node)
+            return true;
+
+        // Allow U-turns at dead-ends
+        if (graph.GetAdjacentEdgeRange(intersection_node).size() == 1)
+            return true;
+
+        // Allow U-turns at dead-ends if there is at most one bidirectional road at the intersection
+        // The condition allows U-turns d→a→d and c→b→c ("Bike - Around the Block" test)
+        //   a→b
+        //   ↕ ↕
+        //   d↔c
+        if (allowed_exits == 1 || bidirectional_edges <= 1)
+            return true;
+
+        // Allow U-turn if the incoming edge has a U-turn lane
+        const auto &incoming_edge_annotation_id = graph.GetEdgeData(from.edge).annotation_data;
```

Comment:
We should probably discuss if using the presence of a `uturn` lane is a good idea here, or if we should base this on country specific settings. Making the accessibility depend on turn lanes is just quite ugly.

In general, we should investigate how often this code path is even triggered - I have a feeling that most streets with a u-turn lane also have segregated ways for both directions (so it is not a "simple" uturn).

Reply:
added a TODO line here, but i would keep it as it is now